larray/core/session.py

```diff
@@ -781,6 +780,50 @@ def compact(self, display=False):
             new_items.append((k, compacted))
         return Session(new_items)
 
+    def apply(self, func, *args, **kwargs):
+        """
+        Apply function `func` on all elements of the session and return a new session.
+
+        Parameters
+        ----------
+        func : function
+            Function to apply to each element of the session. It should take a single `element` argument and return
+            a single value.
+        *args : any
+            Any extra arguments are passed to the function
+        **kwargs : any
+            Any extra keyword arguments are passed to the function
+
+        Returns
+        -------
+        Session
+            A new session containing all processed arrays
+
+        Examples
+        --------
+        >>> arr1 = ndtest(2)
+        >>> arr1
+        a  a0  a1
+            0   1
+        >>> arr2 = ndtest(3)
+        >>> arr2
+        a  a0  a1  a2
+            0   1   2
+        >>> sess1 = Session([('arr1', arr1), ('arr2', arr2)])
+        >>> sess1
+        Session(arr1, arr2)
+        >>> def increment(element):
+        ...     return element + 1
+        >>> sess2 = sess1.apply(increment)
+        >>> sess2.arr1
+        a  a0  a1
+            1   2
+        >>> sess2.arr2
+        a  a0  a1  a2
+            1   2   3
+        """
+        return Session([(k, func(v, *args, **kwargs)) for k, v in self.items()])
```

Comment:
I wouldn't do that, because if you want filter, you can do so using the filter method:
`sess.filter(kind=LArray).apply()`

Oh, now I realize I misread what you wrote. I thought you meant:
`return Session([(k, func(v, *args, **kwargs) if isinstance(v, LArray) else v) for k, v in self.items()])`
maybe having an array=True keyword which apply to arrays only by default would be the best solution?